# HG changeset patch
# User Rik <rik@octave.org>
# Date 1471548392 25200
#      Thu Aug 18 12:26:32 2016 -0700
# Node ID 029c5f4eac3a059f147e7f56d04cbefc10fb42e6
# Parent  609403f90bb7fd443bad892036b6766cdd419d60
rgbplot.m: Don't use Octave-only plot() syntax.

* rgbplot.m: Call plot (x,y,'color', ...) rather than plot (y, 'color', ...)
Use double quotes instead of single quotes.

diff --git a/scripts/image/rgbplot.m b/scripts/image/rgbplot.m
--- a/scripts/image/rgbplot.m
+++ b/scripts/image/rgbplot.m
@@ -24,17 +24,17 @@
 ## Plot the components of a colormap.
 ##
 ## Two different @var{style}s are available for displaying the @var{cmap}:
 ##
 ## @table @asis
 ## @item profile (default)
 ## Plot the RGB line profile of the colormap for each of the channels (red,
 ## green and blue) with the plot lines colored appropriately.  Each line
-## represents the intensity of each RGB components across the colormap.
+## represents the intensity of an RGB component across the colormap.
 ##
 ## @item composite
 ## Draw the colormap across the X-axis so that the actual index colors are
 ## visible rather than the individual color components.
 ##
 ## @end table
 ##
 ## The optional return value @var{h} is a graphics handle to the created plot.
@@ -53,22 +53,23 @@ function h = rgbplot (cmap, style = "pro
   if (! iscolormap (cmap))
     error ("rgbplot: CMAP must be a valid colormap");
   elseif (! ischar (style))
     error ("rgbplot: STYLE must be a string");
   endif
 
   switch (tolower (style))
     case "profile"
-      htmp = plot (cmap(:,1),"r", cmap(:,2),"g", cmap(:,3),"b");
-      set (gca (), 'ytick', 0:0.1:1);
-      set (gca (), 'xlim', [0 rows(cmap)]);
+      x = 1:rows (cmap);
+      htmp = plot (x,cmap(:,1),"r", x,cmap(:,2),"g", x,cmap(:,3),"b");
+      set (gca (), "ytick", 0:0.1:1);
+      set (gca (), "xlim", [0 rows(cmap)]);
     case "composite"
-      htmp = image (1:rows(cmap));
-      set (gca, 'ytick', []);
+      htmp = image (1:rows (cmap));
+      set (gca, "ytick", []);
       colormap (cmap);
     otherwise
       error ("rgbplot: unknown style '%s'", style);
   endswitch
   xlabel ("color index");
 
   if (nargout > 0)
     h = htmp;
